identity-iota/src/credential/errors.rs

```diff
@@ -0,0 +1,138 @@
+  // flexibility to include more information in a simple String? Can the `did_url` be problematic in terms of GDPR if
+  // it gets written to a log file?
+  DeactivatedSubjectDocument { did_url: IotaDIDUrl },
+  /// Indicates that the credential's signature could not be verified using the issuer's DID Document.
+  #[error("could not verify the issuer's signature")]
+  IssuerProof {
+    source: Box<dyn std::error::Error + Send + Sync + 'static>, /* Todo: Would it be better to use a specific type
+                                                                 * here? */
+  },
+  /// Indicates an attempt to validate a credential signed by an untrusted issuer.
+  #[error("the credential is signed by an untrusted issuer")]
+  UntrustedIssuer,
+
+  /// Indicates that the credential's issuer could not be parsed as a valid DID.
+  #[error("the credential's issuer property could not be parsed to a valid DID")]
+  IssuerUrl {
+    source: Box<dyn std::error::Error + Send + Sync + 'static>, /* Todo: Would it be better to use a specific type
+                                                                 * here? */
+  },
+
+  /// Indicates that the credential's issuer could not be parsed as a valid DID.
+  #[error("the presentation's holder property could not be parsed to a valid DID")]
+  HolderUrl {
+    source: Box<dyn std::error::Error + Send + Sync + 'static>, /* Todo: Would it be better to use a specific type
+                                                                 * here? */
+  },
+
+  /// Indicates an attempt to validate a presentation using a resolved DID document not corresponding to the URL of the
+  /// presentation's holder property.
+  #[error("the provided holder document does not correspond to the presentation's holder property")]
+  IncompatibleHolderDocument,
+
+  /// Indicates that the presentation's signature could not be verified using the holder's DID Document.
+  #[error("presentation validation failed: could not verify the holder's signature")]
+  HolderProof {
+    source: Box<dyn std::error::Error + Send + Sync + 'static>, /* Todo: Would it be better to use a specific type
+                                                                 * here? */
+  },
+  /// Indicates that the structure of the [identity_credential::credential::Credential] is not semantically correct.
+  #[error("the credential's structure is not semantically correct")]
+  CredentialStructure(#[source] identity_credential::Error),
+  /// Indicates that the structure of the [identity_credential::presentation::Presentation] is not semantically
+  /// correct.
+  #[error("the presentation's structure is not spec compliant")]
+  PresentationStructure(#[source] identity_credential::Error),
+  /// Indicates that the presentation does not comply with the nonTransferable property of one of its credentials.
+  #[error(
+    "the nonTransferable property of the credential at position {credential_position} in the presentation is not met"
```

Comment:
"Also feels a bit too formal yet very descriptive. Maybe we can add a quick hint "expected holder = subject".

Reply:
Updated the description.